🐛 BUG: Type error:  Cannot find module `../core/build/types`

### What version of `astro` are you using?

1.0.0-beta.27

### Are you using an SSR adapter? If so, which one?

Netlify

### What package manager are you using?

pnpm

### What operating system are you using?

Linux POP!_os 22.04

### Describe the Bug

when I run `tsc -d` to emit type definitions I get an error from a file within node_modules:

```bash
node_modules/astro/dist/types/@types/astro.d.ts:11:36 - error TS2307: Cannot find module '../core/build/types' or its corresponding type declarations.

11 import type { PageBuildData } from '../core/build/types';
                                      ~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/astro/dist/types/@types/astro.d.ts:11
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-vnnnud?file=tsconfig.json&view=editor

### Participation

- [X] I am willing to submit a pull request for this issue.

--- a/.changeset/new-carrots-cough.md
+++ b/.changeset/new-carrots-cough.md
@@ -1,0 +1,5 @@
+---
+"astro": patch
+---
+
+Fix an `import from '../core/build/types';` error

--- a/packages/astro/src/core/build/types.ts
+++ b/packages/astro/src/core/build/types.ts
@@ -7,9 +7,9 @@ import type {
 	ComponentInstance,
 	SSRLoadedRenderer,
 } from '../../@types/astro';
-import type { ViteConfigWithSSR } from '../../create-vite';
-import type { LogOptions } from '../../logger';
-import type { RouteCache } from '../../render/route-cache.js';
+import { LogOptions } from '../logger/core';
+import { RouteCache } from '../render/route-cache';
+import { ViteConfigWithSSR } from '../create-vite';
 
 export type ComponentPath = string;
 export type ViteID = string;

--- a/packages/astro/src/core/build/types.ts
+++ b/packages/astro/src/core/build/types.ts
@@ -7,9 +7,9 @@ import type {
 	ComponentInstance,
 	SSRLoadedRenderer,
 } from '../../@types/astro';
-import { LogOptions } from '../logger/core';
-import { RouteCache } from '../render/route-cache';
-import { ViteConfigWithSSR } from '../create-vite';
+import type { LogOptions } from '../logger/core';
+import type { RouteCache } from '../render/route-cache';
+import type { ViteConfigWithSSR } from '../create-vite';
 
 export type ComponentPath = string;
 export type ViteID = string;

